services/mlmodel/server.go

```diff
@@ -28,24 +30,41 @@ func (server *serviceServer) Infer(ctx context.Context, req *pb.InferRequest) (*
 	if err != nil {
 		return nil, err
 	}
-	id, err := asMap(req.InputData)
+	var id map[string]interface{}
+	if req.InputData != nil {
+		id, err = asMap(req.InputData)
+		if err != nil {
+			return nil, err
+		}
+	}
+	var it ml.Tensors
+	if req.InputTensors != nil {
+		it, err = protoToTensors(req.InputTensors)
```

Comment:
Do we not expect users/devs to have to do this (protoToTensors)? Feels slightly weird to me for one way to be exported but not the other

Reply:
made it public